google-cloud-spanner/src/main/java/com/google/cloud/spanner/connection/ConnectionOptions.java

```diff
@@ -483,18 +496,28 @@ private ConnectionOptions(Builder builder) {
         (builder.credentials == null && this.credentialsUrl == null) || this.oauthToken == null,
         "Cannot specify both credentials and an OAuth token.");
 
-    this.usePlainText = parseUsePlainText(this.uri);
     this.userAgent = parseUserAgent(this.uri);
     QueryOptions.Builder queryOptionsBuilder = QueryOptions.newBuilder();
     queryOptionsBuilder.setOptimizerVersion(parseOptimizerVersion(this.uri));
     this.queryOptions = queryOptionsBuilder.build();
     this.returnCommitStats = parseReturnCommitStats(this.uri);
+    this.autoConfigEmulator = parseAutoConfigEmulator(this.uri);
+    this.usePlainText = this.autoConfigEmulator || parseUsePlainText(this.uri);
+
+    if (matcher.group(Builder.HOST_GROUP) == null) {
```

Comment:
nit: extract method for the `if` `else` block

Reply:
Done